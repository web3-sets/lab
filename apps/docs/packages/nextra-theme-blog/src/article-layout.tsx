import type { ReactNode } from 'react'
import Meta from './meta'
import { MDXTheme } from './mdx-theme'
import { useBlogContext } from './blog-context'
import { BasicLayout } from './basic-layout'

export const ArticleLayout = ({ children }: { children: ReactNode }) => {
  const { config } = useBlogContext()
  return (
    <BasicLayout>
      <Meta />
      <MDXTheme>
        {children}
        {config.postFooter}
        {config.comments}
      </MDXTheme>
    </BasicLayout>
  )
}
