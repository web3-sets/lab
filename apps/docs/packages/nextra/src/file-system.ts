import { findPagesDir, existsSync } from 'next/dist/lib/find-pages-dir.js'
import { CWD } from './constants'

export { existsSync }

export function findPagesDirectory(): string {
  const res = findPagesDir(CWD, false)
  return (
    res.pagesDir || // next v13
    (res as any).pages // next v12
  )
}

export const PAGES_DIR = process.env.VITEST_WORKER_ID
  ? ''
  : findPagesDirectory()
