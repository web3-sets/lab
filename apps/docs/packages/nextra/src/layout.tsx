import type { ReactElement } from 'react'

import { SSGContext } from './ssg'
import { useInternals } from './use-internals'

export default function Nextra({
  __nextra_pageMap,
  __nextra_dynamic_opts,
  ...props
}: any): ReactElement {
  const { context, Layout } = useInternals()
  const { Content, ...restContext } = context

  if (__nextra_pageMap) {
    restContext.pageOpts = {
      ...restContext.pageOpts,
      pageMap: __nextra_pageMap
    }
  }

  if (__nextra_dynamic_opts) {
    const data = JSON.parse(__nextra_dynamic_opts)
    restContext.pageOpts = {
      ...restContext.pageOpts,
      headings: data.headings,
      title: data.title || restContext.pageOpts.title,
      frontMatter: data.frontMatter
    }
  }

  return (
    <Layout {...restContext} pageProps={props}>
      <SSGContext.Provider value={props}>
        <Content {...props} />
      </SSGContext.Provider>
    </Layout>
  )
}
