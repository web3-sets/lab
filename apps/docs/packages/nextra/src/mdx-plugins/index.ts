export { parseMeta, attachMeta } from './rehype'
export { remarkCustomHeadingId } from './remark-custom-heading-id'
export { remarkHeadings } from './remark-headings'
export { remarkReplaceImports } from './remark-replace-imports'
export { remarkRemoveImports } from './remark-remove-imports'
export {
  remarkLinkRewrite,
  type RemarkLinkRewriteOptions
} from './remark-link-rewrite'
export { remarkStaticImage } from './static-image'
export { structurize } from './structurize'
