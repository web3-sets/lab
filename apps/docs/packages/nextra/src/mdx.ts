export { MDXProvider, useMDXComponents } from '@mdx-js/react'
export type { Components } from '@mdx-js/react/lib'
