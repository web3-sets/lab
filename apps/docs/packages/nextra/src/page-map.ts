import type {
  FileMap,
  MdxPath,
  PageMapItem,
  DynamicMetaDescriptor
} from './types'
import filterRouteLocale from './filter-route-locale'
import { IS_PRODUCTION } from './constants'

type PageMapProps = {
  filePath: string
  items: PageMapItem[]
  fileMap: FileMap
  defaultLocale: string
}

export function getDynamicMeta(
  path: string,
  items: PageMapItem[],
  locale?: string
): [DynamicMetaDescriptor[], PageMapItem[]] {
  const newItems: PageMapItem[] = []
  const dynamicMetaItems: DynamicMetaDescriptor[] = []

  for (let i = 0; i < items.length; i++) {
    const item = items[i]
    if (item.kind === 'Folder') {
      const [dynamicItemsInChildren, newItemsInChildren] = getDynamicMeta(
        `${path}[${i}].children`,
        item.children,
        locale
      )
      dynamicMetaItems.push(...dynamicItemsInChildren)
      if (newItemsInChildren.length) {
        newItems.push({
          ...item,
          children: newItemsInChildren
        })
      }
      continue
    }

    if (item.kind === 'Meta' && item.__nextra_src) {
      if (locale === item.locale) {
        const { __nextra_src, ...newItem } = item
        dynamicMetaItems.push({
          metaFilePath: __nextra_src,
          metaObjectKeyPath: `${path}[${i}]`,
          metaParentKeyPath: path.replace(/\.children$/, '')
        })
        newItems.push(newItem)
      }
      continue
    }
    newItems.push(item)
  }

  return [dynamicMetaItems, newItems]
}

const cachedDynamicMetaForLocale: Record<
  string,
  [DynamicMetaDescriptor[], PageMapItem[]]
> = Object.create(null)

export function resolvePageMap({
  filePath,
  items,
  fileMap,
  defaultLocale
}: PageMapProps): {
  route: string
  pageMap: PageMapItem[]
  dynamicMetaItems: DynamicMetaDescriptor[]
} {
  const { locale, route } = fileMap[filePath as MdxPath]
  const localeKey = locale || ''

  const [dynamicMetaItems, newItems] =
    (IS_PRODUCTION && cachedDynamicMetaForLocale[localeKey]) ||
    (cachedDynamicMetaForLocale[localeKey] = getDynamicMeta(
      '',
      locale ? filterRouteLocale(items, locale, defaultLocale) : items,
      locale
    ))

  return {
    pageMap: newItems,
    route,
    dynamicMetaItems
  }
}
