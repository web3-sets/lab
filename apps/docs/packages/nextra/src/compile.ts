import path from 'node:path'
import { createRequire } from 'node:module'
import type { ProcessorOptions } from '@mdx-js/mdx'
import { createProcessor } from '@mdx-js/mdx'
import type { Processor } from '@mdx-js/mdx/lib/core'
import remarkGfm from 'remark-gfm'
import type { Options as RehypePrettyCodeOptions } from 'rehype-pretty-code'
import rehypePrettyCode from 'rehype-pretty-code'
import remarkReadingTime from 'remark-reading-time'
import grayMatter from 'gray-matter'
import remarkMath from 'remark-math'
import rehypeKatex from 'rehype-katex'
import type { Pluggable } from 'unified'

import {
  remarkStaticImage,
  remarkCustomHeadingId,
  remarkHeadings,
  remarkReplaceImports,
  structurize,
  parseMeta,
  attachMeta,
  remarkRemoveImports,
  remarkLinkRewrite
} from './mdx-plugins'
import type { LoaderOptions, PageOpts, ReadingTime } from './types'
import { truthy } from './utils'
import {
  CODE_BLOCK_FILENAME_REGEX,
  CWD,
  DEFAULT_LOCALE,
  MARKDOWN_URL_EXTENSION_REGEX
} from './constants'
import theme from './theme.json'

globalThis.__nextra_temp_do_not_use = () => {
  import('./__temp__')
}

const require = createRequire(import.meta.url)

const DEFAULT_REHYPE_PRETTY_CODE_OPTIONS: RehypePrettyCodeOptions = {
  // @ts-expect-error -- TODO: fix type error
  theme,
  onVisitLine(node: any) {
    // Prevent lines from collapsing in `display: grid` mode, and
    // allow empty lines to be copy/pasted
    if (node.children.length === 0) {
      node.children = [{ type: 'text', value: ' ' }]
    }
  },
  onVisitHighlightedLine(node: any) {
    node.properties.className.push('highlighted')
  },
  onVisitHighlightedWord(node: any) {
    node.properties.className = ['highlighted']
  },
  filterMetaString: (meta: string) =>
    meta.replace(CODE_BLOCK_FILENAME_REGEX, '')
}

const cachedCompilerForFormat: Record<
  Exclude<ProcessorOptions['format'], undefined | null>,
  Processor
> = Object.create(null)

type MdxOptions = LoaderOptions['mdxOptions'] &
  Pick<ProcessorOptions, 'jsx' | 'outputFormat'>

export async function compileMdx(
  source: string,
  loaderOptions: Pick<
    LoaderOptions,
    | 'staticImage'
    | 'flexsearch'
    | 'defaultShowCopyCode'
    | 'readingTime'
    | 'latex'
    | 'codeHighlight'
  > & { mdxOptions?: MdxOptions; route?: string; locale?: string } = {},
  { filePath = '', useCachedCompiler = false, isPageImport = true } = {}
) {
  // Extract frontMatter information if it exists
  const { data: frontMatter, content } = grayMatter(source)

  const structurizedData = Object.create(null)

  let searchIndexKey: string | null = null
  if (typeof loaderOptions.flexsearch === 'object') {
    if (loaderOptions.flexsearch.indexKey) {
      searchIndexKey = loaderOptions.flexsearch.indexKey(
        filePath,
        loaderOptions.route || '',
        loaderOptions.locale
      )
      if (searchIndexKey === '') {
        searchIndexKey = loaderOptions.locale || DEFAULT_LOCALE
      }
    } else {
      searchIndexKey = loaderOptions.locale || DEFAULT_LOCALE
    }
  } else if (loaderOptions.flexsearch) {
    searchIndexKey = loaderOptions.locale || DEFAULT_LOCALE
  }

  const {
    jsx = false,
    format: _format = 'mdx',
    outputFormat = 'function-body',
    remarkPlugins,
    rehypePlugins,
    rehypePrettyCodeOptions
  }: MdxOptions = {
    ...loaderOptions.mdxOptions,
    // You can override MDX options in the frontMatter too.
    ...frontMatter.mdxOptions
  }

  const format =
    _format === 'detect' ? (filePath.endsWith('.mdx') ? 'mdx' : 'md') : _format

  const {
    staticImage,
    flexsearch,
    readingTime,
    latex,
    codeHighlight,
    defaultShowCopyCode
  } = loaderOptions

  // https://github.com/shuding/nextra/issues/1303
  const isFileOutsideCWD =
    !isPageImport && path.relative(CWD, filePath).startsWith('..')

  const compiler =
    (useCachedCompiler && cachedCompilerForFormat[format]) ||
    (cachedCompilerForFormat[format] = createProcessor({
      jsx,
      format,
      outputFormat,
      providerImportSource: isFileOutsideCWD
        ? require.resolve('nextra').replace(/index\.js$/, 'mdx.mjs') // fixes Package subpath './mdx' is not defined by "exports"
        : 'nextra/mdx',
      remarkPlugins: [
        ...(remarkPlugins || []),
        outputFormat === 'function-body' && remarkRemoveImports,
        remarkGfm,
        remarkCustomHeadingId,
        remarkHeadings,
        staticImage && remarkStaticImage,
        searchIndexKey !== null && structurize(structurizedData, flexsearch),
        readingTime && remarkReadingTime,
        latex && remarkMath,
        isFileOutsideCWD && remarkReplaceImports,
        // Remove the markdown file extension from links
        [
          remarkLinkRewrite,
          {
            pattern: MARKDOWN_URL_EXTENSION_REGEX,
            replace: '',
            excludeExternalLinks: true
          }
        ] satisfies Pluggable
      ].filter(truthy),
      rehypePlugins: [
        ...(rehypePlugins || []),
        [parseMeta, { defaultShowCopyCode }],
        codeHighlight !== false &&
          ([
            rehypePrettyCode,
            {
              ...DEFAULT_REHYPE_PRETTY_CODE_OPTIONS,
              ...rehypePrettyCodeOptions
            }
          ] as any),
        attachMeta,
        latex && rehypeKatex
      ].filter(truthy)
    }))

  try {
    compiler.data('headingMeta', { headings: [] })
    const vFile = await compiler.process(
      filePath ? { value: content, path: filePath } : content
    )

    const headingMeta = compiler.data('headingMeta') as Pick<
      PageOpts,
      'headings' | 'hasJsxInH1' | 'title'
    >
    const readingTime = vFile.data.readingTime as ReadingTime | undefined

    return {
      // https://github.com/shuding/nextra/issues/1032
      result: String(vFile).replaceAll('__esModule', '_\\_esModule'),
      ...headingMeta,
      ...(readingTime && { readingTime }),
      structurizedData,
      searchIndexKey,
      frontMatter
    }
  } catch (err) {
    console.error(`[nextra] Error compiling ${filePath}.`)
    throw err
  }
}
