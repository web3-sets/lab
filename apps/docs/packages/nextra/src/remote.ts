import { compileMdx } from './compile'
import type { RemarkLinkRewriteOptions } from './mdx-plugins'
import { remarkLinkRewrite } from './mdx-plugins'
import { truthy } from './utils'

export const buildDynamicMDX = async (
  content: string,
  {
    remarkLinkRewriteOptions,
    ...loaderOptions
  }: Parameters<typeof compileMdx>[1] & {
    remarkLinkRewriteOptions?: RemarkLinkRewriteOptions
  } = {}
) => {
  const { result, headings, frontMatter, title } = await compileMdx(content, {
    ...loaderOptions,
    mdxOptions: {
      ...loaderOptions.mdxOptions,
      remarkPlugins: [
        ...(loaderOptions.mdxOptions?.remarkPlugins || []),
        remarkLinkRewriteOptions &&
          ([remarkLinkRewrite, remarkLinkRewriteOptions] as any)
      ].filter(truthy)
    }
  })

  return {
    __nextra_dynamic_mdx: result,
    __nextra_dynamic_opts: JSON.stringify({
      headings,
      frontMatter,
      title: frontMatter.title || title
    })
  }
}

export const buildDynamicMeta = async () => {
  const resolvePageMap = (globalThis as any).__nextra_resolvePageMap
  if (resolvePageMap) {
    return {
      __nextra_pageMap: await resolvePageMap()
    }
  }
  return {}
}
