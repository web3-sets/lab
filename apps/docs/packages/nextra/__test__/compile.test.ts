import fs from 'node:fs/promises'
import path from 'node:path'
import { compileMdx } from '../src/compile'
import { it, describe, expect } from 'vitest'
import { CWD } from '../src/constants'

type Name =
  | 'code-h1.mdx'
  | 'code-with-text-h1.mdx'
  | 'dynamic-h1.mdx'
  | 'no-h1.mdx'
  | 'static-h1.mdx'

function loadFixture(name: Name) {
  const filePath = path.join(CWD, '__test__', 'fixture', 'headings', name)
  return fs.readFile(filePath, 'utf8')
}

const mdxOptions = {
  jsx: true,
  outputFormat: 'program' as const
}

describe('Process heading', () => {
  it('code-h1', async () => {
    const data = await loadFixture('code-h1.mdx')
    const result = await compileMdx(data, { mdxOptions })
    expect(result).toMatchSnapshot()
  })
  it('code-with-text-h1', async () => {
    const data = await loadFixture('code-with-text-h1.mdx')
    const result = await compileMdx(data, { mdxOptions })
    expect(result).toMatchSnapshot()
  })
  it('static-h1', async () => {
    const data = await loadFixture('static-h1.mdx')
    const result = await compileMdx(data, { mdxOptions })
    expect(result).toMatchSnapshot()
  })
  it('dynamic-h1', async () => {
    const data = await loadFixture('dynamic-h1.mdx')
    const result = await compileMdx(data, { mdxOptions })
    expect(result).toMatchSnapshot()
  })
  it('no-h1', async () => {
    const data = await loadFixture('no-h1.mdx')
    const result = await compileMdx(data, { mdxOptions })
    expect(result).toMatchSnapshot()
  })
  it('use custom heading id', async () => {
    const { result } = await compileMdx(
      `
# My Header [#test-id]
## Some extra space [#extra-space]&nbsp;
### Some extra space in heading    [#extra-space-in-heading]
### nospace[#without-space]
#### foo [#другой язык]
##### bar Baz []
###### bar Qux [#]`,
      { mdxOptions }
    )
    expect(result).toMatch(`<_components.h1 id="test-id">{"My Header"}`)
    expect(result).toMatch(
      `<_components.h2 id="extra-space">{"Some extra space"}</_components.h2>`
    )
    expect(result).toMatch(
      `<_components.h3 id="extra-space-in-heading">{"Some extra space in heading"}`
    )
    expect(result).toMatch(
      `<_components.h3 id="without-space">{"nospace"}</_components.h3>`
    )
    expect(result).toMatch(`<_components.h4 id="другой язык">{"foo"}`)
    expect(result).toMatch(`<_components.h5 id="bar-baz-">{"bar Baz []"}`)
    expect(result).toMatch(`<_components.h6 id="bar-qux-">{"bar Qux [#]"}`)
  })
  it('use github-slugger', async () => {
    const { result } = await compileMdx(`### My Header`, { mdxOptions })
    expect(result).toMatch(`<_components.h3 id="my-header">{"My Header"}`)
  })
})

describe('Link', () => {
  it('supports .md links', async () => {
    const { result } = await compileMdx(`[link](../file.md)`, { mdxOptions })
    expect(result).toMatch(`<_components.a href="../file">`)
  })

  it('supports .mdx links', async () => {
    const { result } = await compileMdx(`[link](../file.mdx)`, { mdxOptions })
    expect(result).toMatch(`<_components.a href="../file">`)
  })

  it('supports URL links', async () => {
    const { result } = await compileMdx(`[link](../file)`, { mdxOptions })
    expect(result).toMatch(`<_components.a href="../file">`)
  })

  it('supports query', async () => {
    const { result } = await compileMdx(`[link](../file.md?query=a)`, {
      mdxOptions
    })
    expect(result).toMatch(`<_components.a href="../file?query=a">`)
  })

  it('supports anchor', async () => {
    const { result } = await compileMdx(`[link](../file.md#anchor)`, {
      mdxOptions
    })
    expect(result).toMatch(`<_components.a href="../file#anchor">`)
  })

  it('supports external .md links', async () => {
    const { result } = await compileMdx(`[link](https://example.com/file.md)`, {
      mdxOptions
    })
    expect(result).toMatch(`<_components.a href="https://example.com/file.md">`)
  })

  it('supports external .mdx links', async () => {
    const { result } = await compileMdx(
      `[link](https://example.com/file.mdx)`,
      {
        mdxOptions
      }
    )
    expect(result).toMatch(
      `<_components.a href="https://example.com/file.mdx">`
    )
  })
})

describe('Code block', () => {
  describe('Filename', () => {
    it('attach with "codeHighlight: true" by default', async () => {
      const { result } = await compileMdx('```js filename="test.js"\n```', {
        mdxOptions
      })
      expect(result).toMatch(
        '<_components.pre data-language="js" data-theme="default" filename="test.js">'
      )
    })

    it('attach with "codeHighlight: false"', async () => {
      const { result } = await compileMdx('```js filename="test.js"\n```', {
        mdxOptions,
        codeHighlight: false
      })
      expect(result).toMatch('<_components.pre filename="test.js">')
    })

    it('not highlight filename as substring', async () => {
      const { result } = await compileMdx('```js filename="/foo/"\nfoo\n```', {
        mdxOptions,
        codeHighlight: true // processed only by rehype-pretty-code
      })
      expect(result).not.toMatch(
        'className="highlighted">{"foo"}</_components.span>'
      )
      expect(result).toMatch('}}>{"foo"}</_components.span>')
    })
  })

  describe('Highlight', () => {
    it('should support line highlights', async () => {
      const { result } = await compileMdx(
        '```js filename="test.js" {1}\n123\n```',
        {
          mdxOptions
        }
      )
      expect(result).toMatch('<_components.span className="line highlighted">')
    })
  })

  describe('Copy code button', () => {
    for (const codeHighlight of [true, false]) {
      describe(`codeHighlight: ${codeHighlight}`, () => {
        it(`attach with "copy"`, async () => {
          const { result } = await compileMdx('```js copy\n```', {
            mdxOptions,
            codeHighlight
          })
          expect(result).toMatch('hasCopyCode>')
        })

        it(`attach with "defaultShowCopyCode: true"`, async () => {
          const { result } = await compileMdx('```js\n```', {
            mdxOptions,
            defaultShowCopyCode: true,
            codeHighlight
          })
          expect(result).toMatch('hasCopyCode>')
        })

        it(`not attach with "defaultShowCopyCode: true" and "copy=false"`, async () => {
          const { result } = await compileMdx('```js copy=false\n```', {
            mdxOptions,
            defaultShowCopyCode: true,
            codeHighlight
          })
          expect(result).not.toMatch('hasCopyCode>')
        })
      })
    }
  })

  it('code block without language has "text" language', async () => {
    const { result } = await compileMdx('```\n```', {
      mdxOptions,
      codeHighlight: false
    })
    expect(result).toMatch('<_components.code className="language-text" />')
  })
})
