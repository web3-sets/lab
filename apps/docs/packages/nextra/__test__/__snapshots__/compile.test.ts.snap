// Vitest Snapshot v1

exports[`Process heading > code-h1 1`] = `
{
  "frontMatter": {},
  "headings": [
    {
      "depth": 1,
      "id": "codegenyml",
      "value": "codegen.yml",
    },
  ],
  "result": "/*@jsxRuntime automatic @jsxImportSource react*/
import {useMDXComponents as _provideComponents} from \\"nextra/mdx\\";
function _createMdxContent(props) {
  const _components = Object.assign({
    h1: \\"h1\\",
    code: \\"code\\"
  }, _provideComponents(), props.components);
  return <_components.h1><_components.code>{\\"codegen.yml\\"}</_components.code></_components.h1>;
}
function MDXContent(props = {}) {
  const {wrapper: MDXLayout} = Object.assign({}, _provideComponents(), props.components);
  return MDXLayout ? <MDXLayout {...props}><_createMdxContent {...props} /></MDXLayout> : _createMdxContent(props);
}
export default MDXContent;
",
  "searchIndexKey": null,
  "structurizedData": {},
  "title": "codegen.yml",
}
`;

exports[`Process heading > code-with-text-h1 1`] = `
{
  "frontMatter": {},
  "headings": [
    {
      "depth": 1,
      "id": "codegenyml-file",
      "value": "codegen.yml file",
    },
  ],
  "result": "/*@jsxRuntime automatic @jsxImportSource react*/
import {useMDXComponents as _provideComponents} from \\"nextra/mdx\\";
function _createMdxContent(props) {
  const _components = Object.assign({
    h1: \\"h1\\",
    code: \\"code\\"
  }, _provideComponents(), props.components);
  return <_components.h1><_components.code>{\\"codegen.yml\\"}</_components.code>{\\" file\\"}</_components.h1>;
}
function MDXContent(props = {}) {
  const {wrapper: MDXLayout} = Object.assign({}, _provideComponents(), props.components);
  return MDXLayout ? <MDXLayout {...props}><_createMdxContent {...props} /></MDXLayout> : _createMdxContent(props);
}
export default MDXContent;
",
  "searchIndexKey": null,
  "structurizedData": {},
  "title": "codegen.yml file",
}
`;

exports[`Process heading > dynamic-h1 1`] = `
{
  "frontMatter": {},
  "hasJsxInH1": true,
  "headings": [
    {
      "depth": 1,
      "id": "posts-tagged-with-",
      "value": "Posts Tagged with “”",
    },
  ],
  "result": "/*@jsxRuntime automatic @jsxImportSource react*/
import {useMDXComponents as _provideComponents} from \\"nextra/mdx\\";
import {useRouter} from 'next/router';
export const TagName = () => {
  const {tag} = useRouter().query;
  return tag || null;
};
function _createMdxContent(props) {
  const _components = Object.assign({
    h1: \\"h1\\"
  }, _provideComponents(), props.components);
  return <_components.h1>{\\"Posts Tagged with “\\"}<TagName />{\\"”\\"}</_components.h1>;
}
function MDXContent(props = {}) {
  const {wrapper: MDXLayout} = Object.assign({}, _provideComponents(), props.components);
  return MDXLayout ? <MDXLayout {...props}><_createMdxContent {...props} /></MDXLayout> : _createMdxContent(props);
}
export default MDXContent;
",
  "searchIndexKey": null,
  "structurizedData": {},
  "title": "Posts Tagged with “”",
}
`;

exports[`Process heading > no-h1 1`] = `
{
  "frontMatter": {},
  "headings": [
    {
      "depth": 2,
      "id": "h2",
      "value": "H2",
    },
  ],
  "result": "/*@jsxRuntime automatic @jsxImportSource react*/
import {useMDXComponents as _provideComponents} from \\"nextra/mdx\\";
function _createMdxContent(props) {
  const _components = Object.assign({
    h2: \\"h2\\"
  }, _provideComponents(), props.components);
  return <_components.h2 id=\\"h2\\">{\\"H2\\"}</_components.h2>;
}
function MDXContent(props = {}) {
  const {wrapper: MDXLayout} = Object.assign({}, _provideComponents(), props.components);
  return MDXLayout ? <MDXLayout {...props}><_createMdxContent {...props} /></MDXLayout> : _createMdxContent(props);
}
export default MDXContent;
",
  "searchIndexKey": null,
  "structurizedData": {},
}
`;

exports[`Process heading > static-h1 1`] = `
{
  "frontMatter": {},
  "headings": [
    {
      "depth": 1,
      "id": "hello-world",
      "value": "Hello World",
    },
  ],
  "result": "/*@jsxRuntime automatic @jsxImportSource react*/
import {useMDXComponents as _provideComponents} from \\"nextra/mdx\\";
function _createMdxContent(props) {
  const _components = Object.assign({
    h1: \\"h1\\"
  }, _provideComponents(), props.components);
  return <_components.h1>{\\"Hello World\\"}</_components.h1>;
}
function MDXContent(props = {}) {
  const {wrapper: MDXLayout} = Object.assign({}, _provideComponents(), props.components);
  return MDXLayout ? <MDXLayout {...props}><_createMdxContent {...props} /></MDXLayout> : _createMdxContent(props);
}
export default MDXContent;
",
  "searchIndexKey": null,
  "structurizedData": {},
  "title": "Hello World",
}
`;
